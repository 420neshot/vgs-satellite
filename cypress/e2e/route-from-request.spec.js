describe('Localhoste route from request flow', function() {
  beforeEach(() => {
    cy.intercept('GET', 'flows.json').as('getFlows');
    cy.intercept('GET', 'route').as('getRoutes');
    cy.intercept('POST', 'route').as('postRoute');
    cy.cleanupFlows();
    cy.cleanupRoutes();
    cy.fixCypressSpec(__filename);
  });

  it('Creates route from request', function() {
    cy.visit('/');
    cy.get('.menu-item').contains('Logs').click();
    cy.wait(['@getFlows']);
    cy.get('[data-role="no-logs"]').toMatchSnapshot();

    cy.task('viaProxy', {
      method: 'POST',
      url: 'http://httpbin.org/post',
      data: { foo: 'bar' },
    });

    cy.get('[data-role="logs-row"] .badge-success').contains('200');
    cy.get('[data-role="logs-row"]').contains('/post').click();

    cy.get('[data-role="log-details-modal"]');
    cy.get('button').contains('Secure this payload').click({ force: true });
    cy.get('[data-role="quick-integration-modal"]').contains('Select the fields that you want to Redact');
    cy.get('[data-role="quick-integration-modal"] input[type="checkbox"]').click({ force: true });

    cy.get('[data-role="select-secure-payload"]').click();
    cy.get('.tab-pane.active button').contains('Save Outbound route').click({ force: true });
    cy.wait('@postRoute');

    cy.get('.menu-item').contains('Routes').click();
    cy.wait('@getRoutes');

    cy.get('.tab-pane.active [data-role="route-item"]');
  });
});
