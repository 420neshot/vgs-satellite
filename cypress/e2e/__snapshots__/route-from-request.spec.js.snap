exports[`Localhoste route from request flow > Creates route from request #0`] = `
<div data-role="no-logs">
  <p class="text-lg mb-2">ğŸ˜³</p>
  <p class="mb-2">
    There are currently no logs<br /><button
      data-role="show-demo-curl"
      type="button"
      class="ant-btn ant-btn-link"
    >
      <span>Send a request</span>
    </button>
  </p>
</div>
`;
